### ISSUE TITLE
What's the difference to jc21/nginx-proxy-manager Docker Container?

### ISSUE THREAD

**Issue by schittli** [jlesage/docker-nginx-proxy-manager#256]
Hello
thank you very much for sharing your work!

I have a question:
There must be a good reason why this project does not invest its efforts directly in NginxProxyManager/nginx-proxy-manager, which also provides a Docker container.

Would it be possible for you to explain it briefly in the introduction at README.md?

10M+ pulls are already very impressive...
If the users understand the difference, then I suspect you will have even more users 😃 

Thanks a lot, kind regards,
Thomas

**Comment by witnessus** [comment]
The only difference is that the jc21's docker nginx listen port is 80 and 443, admin port is 8181.
The docker is suitable for China, because 80 and 443 port is banned in P.R.C without permition.

**Comment by cmccallu** [comment]
If you examine the [Dockerfile](https://github.com/jlesage/docker-nginx-proxy-manager/blob/master/Dockerfile) you will notice quite a few differences to jc21's image e.g. logrotate. I have found this as I have been looking to migrate over and have it up and running and found this additional stuff in this image which is very useful.

Initially I used this image as it defaulted to SQLite and didn't require another container with a DB which the original jc21's image did but seems to be on longer the case.  

More regular updates would be great on this when the upstream gets a new version.

**Comment by rogierlommers** [comment]
Most important is that this repo DOES support passing the guid/uid params, right? Can somebody confirm this really works?

**Comment by jlesage** [comment]
> Initially I used this image as it defaulted to SQLite and didn't require another container with a DB which the original jc21's image did but seems to be no longer the case.

You are right, this was the main reason initially.  The original was requiring usage of a separate container for the database.

Also, with this project, nginx runs as non-root, on non-privileged ports.  The orignal project requires nginx to be run as root.

Finally, the baseimage used to support the Nginx Proxy Manager application is now quite different.

**Comment by jlesage** [comment]
> More regular updates would be great on this when the upstream gets a new version.

You are right :) It took a lot of time for the last version of NPM to be integrated mainly because I was working on the next major version of the baseimage.  Now that this is done, integrating new NPM versions should be quicker.

**Comment by jlesage** [comment]
> Most important is that this repo DOES support passing the guid/uid params, right? Can somebody confirm this really works?

Yes it does!

**Comment by cmccallu** [comment]
Could be useful to detail all the additional features you have in your container?

**Comment by V-ed** [comment]
As of this commit : https://github.com/NginxProxyManager/nginx-proxy-manager/commit/9fe07fa6c328a4e756480adffc19ca8ed82da1ff , the official image also added in its official documentation the ability to run the image as a specified user/group.

> The docker is suitable for China, because 80 and 443 port is banned in P.R.C without permition.

I also find this a bit moot as a point as you can simply use custom ports when setting up the official docker, here's a quick example (view the `ports` section) :

```dockerfile
version: '3.8'
services:
  app:
    image: 'jc21/nginx-proxy-manager:latest'
    restart: unless-stopped
    ports:
      # These ports are in format <host-port>:<container-port>
      - '8080:80' # Public HTTP Port
      - '4443:443' # Public HTTPS Port
      - '8181:81' # Admin Web Port
```

Even the default database is now SQLite...

---

With the current state of the official image, is there anything else (noteworthy) that using this docker image that makes it different than the official one?

_By "noteworthy", for an counter-example, I would dismiss that this image uses a single volume for all its data as you can simply put both official volumes into a single folder._

I'm currently wondering if I should switch to the official as this would mean more timely updates (_even though you did do a great, if not excellent, job keeping this repo up, I don't want to dismiss the work you did here!_).
Thanks!

**Comment by wenyuwen2011** [comment]
it is quite simple to have a quick start for me as a fresh man. I have difficulty to install docker and docker-compose. I don't need to do so with this image.
